packages/datadog-plugin-mongodb-core/src/index.js

```diff
@@ -90,11 +89,28 @@ function sanitizeBigInt (data) {
   return JSON.stringify(data, (_key, value) => typeof value === 'bigint' ? value.toString() : value)
 }
 
+function extractQuery (statements) {
+  if (statements.length === 1) return statements[0].q
+
+  const extractedQueries = []
+  for (let i = 0; i < statements.length; i++) {
+    const statement = statements[i]
+    extractedQueries.push(statement.q)
```

Comment:
Instead of calling `limitDepth()` on what this function returns, you can just call it on every item you add to `extractedQueries`. That way you don't have to loop through them twice (once here, and once in `limitDepth()`), and you don't need to change `limitDepth()`.